lib/v2/screens/sign_up/claim_invite_screen.dart

```diff
@@ -0,0 +1,190 @@
+        borderRadius: BorderRadius.only(
+          topLeft: Radius.circular(18),
+          topRight: Radius.circular(18),
+        ),
+      ),
+      child: Padding(
+        padding: const EdgeInsets.symmetric(horizontal: 20, vertical: 30),
+        child: Column(
+          mainAxisSize: MainAxisSize.max,
+          crossAxisAlignment: CrossAxisAlignment.start,
+          children: [
+            Text(
+              'Or enter by yourself below',
+              style: Theme.of(context).accentTextTheme.subtitle2,
+            ),
+            const SizedBox(height: 16),
+            TextFormField(
+              controller: _keyController,
+              style: Theme.of(context).accentTextTheme.bodyText1,
+              onChanged: _onInviteCodeChanged,
+              decoration: InputDecoration(
+                hintText: 'Invite code (5 words)'.i18n,
+                hintStyle: Theme.of(context).accentTextTheme.bodyText1,
+                errorText: _currentState.pageState == PageState.failure
+                    ? _currentState.errorMessage
+                    : null,
+                suffixIcon: _inviteCodeSuffixIcon,
+                enabledBorder: const OutlineInputBorder(
+                  borderRadius: BorderRadius.all(Radius.circular(8)),
+                  borderSide: BorderSide(
+                    width: 1,
+                    color: Colors.black,
+                  ),
+                ),
+                focusedBorder: const OutlineInputBorder(
+                  borderRadius: BorderRadius.all(Radius.circular(8)),
+                  borderSide: BorderSide(
+                    width: 1,
+                    color: Colors.black,
+                  ),
+                ),
+                errorBorder: const OutlineInputBorder(
+                  borderRadius: BorderRadius.all(Radius.circular(8)),
+                  borderSide: BorderSide(
+                    width: 1,
+                    color: AppColors.red,
+                  ),
+                ),
+              ),
+            ),
+            const SizedBox(height: 50),
+            FlatButtonLong(
+              title: "Claim code".i18n,
+              onPressed: _onClaimPressed(context),
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+
+  Widget get _inviteCodeSuffixIcon {
+    return _currentState.pageState == PageState.success &&
+            _currentState.inviteModel != null
+        ? const Icon(
+            Icons.check_circle,
+            color: AppColors.green,
+          )
+        : IconButton(
+            icon: const Icon(
+              Icons.paste,
+              color: AppColors.grey,
+            ),
+            onPressed: () async {
+              var clipboardData = await Clipboard.getData('text/plain');
+              var clipboardText = clipboardData?.text ?? '';
+              setState(() {
+                _keyController.text = clipboardText;
+              });
+            },
+          );
+  }
```

Comment:
Refactor portions of the widget tree into widget classes instead of methods, getters ... whatever. For optimization reasons and to separate these components of the screen in another folder.

_buildBottomContainer and _inviteCodeSuffixIcon should be classes and put in the components folder.

If you need to access the state in these components use a Blocbuilder or just the Provider.of <MyBloc> (context) .state.myVariable if you don't want to redraw

Reply:
Addressed